README.md

```diff
@@ -1,54 +1,201 @@
-> Notes:
-> - The ```NodeNetworkState`` CRD has an "un-managed" indicator, allowing an administrator to stop all enforcement and reporting for a node.
-> - The desire state could be created based on ```NodeNetConfPolicy``` CRDs (see below), or just manually set by an external system.
-### Policy Handler
-Can run in distributed or centralized mode. In case of distributed (default mode), it will only handle the ```NodeNetworkState``` CRDs of node it is executed on. In case of centralized mode, there has to be only one instance of it that run at the same time.
-#### Distributed Mode
-Upon invocation, it reads the list of ```NodeNetworkState``` CRDs, as well as the list of ```NodeNetConfPolicy``` CRDs. It will find which ```NodeNetworkState``` CRD is for the node it is running on. It will also find all ```NodeNetConfPolicy``` CRDs that apply to that node (based on their [affinity and toleration](https://kubernetes.io/docs/concepts/configuration/assign-pod-node/#affinity-and-anti-affinity) objects). 
-From the interface match logic in the applicable ```NodeNetConfPolicy``` CRDs, and the list of interfaces taken from the current state of the node (in the ```NodeNetworkState``` CRD), it will create aggregated desired state object, and update it into the relevant ```NodeNetworkState``` CRD.
-When running in "client" mode, it has nothing more to do. When running as a daemon, it will:
-- Detects updates to ```NodeNetConfPolicy``` CRDs applicable for current node, and update ```NodeNetworkState``` CRD if needed.
-- Detect an update to the current state of ```NodeNetworkState``` CRD for the node, and see if its desired state needs to be modified.
-#### Centralized mode
-Very similar to the distributed mode, but in this case, the client od daemon must handle policies and states for all nodes in the system.
-## Getting Started
-To deploy kubernetes-nmstate on the cluster call:
-```
-kubectl --kubeconfig ~/.kube/config apply -f manifests/deployment.yaml
-```
-This would install the CRDs, the state and policy daemon sets, and RBACs. Then call:
+
+We use [nmstate](https://nmstate.github.io/) to perform state driven network
+configuration on each node, as well as to report back its current state.
+The system defines 2 objects, `NodeNetworkState` and `NodeNetConfPolicy` that
+are used to report and configure cluster node interfaces.
+
+You can find more information in [kubernetes-nmstate design document](https://docs.google.com/document/d/1282BcYjYGIIxQKgMYi3nQodB4ML_gw9BSs5AXB7QUtg/) and in [developer guide](docs/developer-guide.md).
+
+### Node Network State
+
+Node Network State objects are created per each node in the cluster and can be
+used both to report available interfaces and to configure them.
+
+Read `NodeNetworkState` of node `node01` to list available interfaces and their
```

Comment:
Please mention that this is an example sub-section.

Reply:
Done